trunk/modules/python/dataset/reader.py

```diff
@@ -78,53 +83,109 @@ def openSICD(image, log=None) :
     wbData, cmplxData = pysix.six_sicd.read(image, schemaPaths)
     return (wbData, cmplxData)
 
-def readSICD(image, log=None) :
+def readSICD(image, log=None, **kwargs) :
     '''This method should read and prepare the data for training or testing.'''
     wbData, cmplxData = openSICD(image, log)
-    return convertPhaseAmp(wbData, log)
+    raw = kwargs.get('raw', False)
+    if not raw:
+        wbData = convertPhaseAmp(wbData, log)
+    else:
+        wbData = wbData, 'sicd'
+    return wbData
 
-def readSIDD(image, log=None) :
+def readSIDD(image, log=None, **kwargs) :
     '''This method should read a prepare the data for training or testing.'''
     raise NotImplementedError('Implement the datasetUtils.readSIDD() method')
 
 def openSIO(image, log=None) :
     import coda.sio_lite
+    if type(image) == np.ndarray:
+        return image
     return coda.sio_lite.read(image)
 
-def readSIO(image, log=None) :
-    imData = openSIO(image, log)
+def transformSIOData(imData, log=None):
```

Comment:
This logic should be added to readImage() and perfomed after the open* methods are called.